packages/core/src/claim/Claim.ts

```diff
@@ -151,8 +156,16 @@ export default class Claim implements IClaim {
    * @returns A new [[Claim]] object.
    */
 
-  public static decompress(compressedClaim: CompressedClaim): Claim {
-    const decompressedClaim = ClaimUtils.decompress(compressedClaim)
+  public static decompress(
+    compressedClaim: CompressedClaim
+  ): Claim | PartialClaim {
```

Comment:
For typechecking the `Claim` type has no effect, but it's nice as a hint for the consumer

Reply:
you could work with overloads as in ClaimUtils

Reply:
just thinking, a function that either returns an instantiated class instance or just some general object feels a little tricky to use. What do we need it for?

Reply:
Maybe, separating the two would be an option.

Reply:
Another option would be not creating a claim. Either returning a full claim or partial claim. Then the user can handle as they need.

Reply:
I would prefer the second option.

Reply:
that would be aliasing the decompress function in the claim.utils, right? Because that's what that one does. I would also prefer that option.

Reply:
Yes, I have updated this as well.